### PR TITLE
[FLINK-13517][docs][hive] Restructure Hive Catalog documentation

### PR DESCRIPTION
**Pull request by sjwiesman**
## What is the purpose of the change

Hive documentation is currently spread across a number of pages and fragmented. In particular: 

1) An example was added to getting-started/examples, however, this section is being removed
2) There is a dedicated page on hive integration but also a lot of hive specific information is on the catalog page

## Brief change log

1) Inline the example into the hive integration page
2) Move the hive specific information on catalogs.md to hive_integration.md
3) Make catalogs.md be just about catalogs in general and link to the hive integration. 
4) Fix some broken links

## Verifying this change

This change is a trivial rework

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable /** docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by sjwiesman** [comment]
@xuefuz would you like to take a look at this. I didn't really change anything, this pr mostly just moves things around.

**Comment by xuefuz** [comment]
Hi @sjwiesman , thanks for bringing up the issue and providing a fix. Your changes look good to me. Nevertheless, Hive documentation needs more refinement including https://issues.apache.org/jira/browse/FLINK-13501.  We will revisit the whole doc and add additional fixes as needed. Thanks for reorganizing the content.

**Comment by sjwiesman** [comment]
Thanks for looking so quickly and sorry about that, I hadn't seen the other ticket. I'm happy to work with you on further refinements.

**Comment by xuefuz** [comment]
@sjwiesman The CI passed. Should we merge this PR?

**Comment by sjwiesman** [comment]
Thank you for the reminder, I am not a committer. @fhueske  or @bowenli86 could one of you please merge this.

**Comment by xuefuz** [comment]
@sjwiesman Would you like to respond to @bowenli86 's comment above? In either case, the PR seems needing update due to conflicts. Thanks.

**Comment by sjwiesman** [comment]
@bowenli86 I've responded to your comment on the JIRA 
@xuefuz I will rebase once we have decided on next steps on JIRA

**Comment by sjwiesman** [comment]
I have pushed an update that I believe addresses everyone's concerns.

File: docs/dev/table/catalog.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-title: "Catalog"
+title: "Catalogs"
```
Comment (bowenli86) [inline docs/dev/table/catalog.md]: update the file name along with such change?

File: docs/dev/table/hive/index.md [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+Flink offers a two-fold integration with Hive. The first is to leverage Hive's Metastore as a persistent catalog for storing Flink specific metadata across sessions. The second is to offer Flink as an alternative engine for reading and writing Hive tables.
+
+The hive catalog is designed to be “out of the box” compatible with existing Hive installations. You do not need to modify your existing Hive Metastore or change the data placement or partitioning of your tables.
+
+* This will be replaced by the TOC
+{:toc}
+
+## Supported Hive Version's
+
+Flink supports Hive `2.3.4` and `1.2.1` and relies and Hive's compatability guarantee's for other versions.
```
Comment (bowenli86) [inline docs/dev/table/hive/index.md]:
```suggestion
Flink supports Hive `2.3.4` and `1.2.1` and relies on Hive's compatibility guarantee's for other versions.
```

File: docs/dev/table/hive/index.zh.md [hunk 3]
```diff
@@ -0,0 +1,261 @@
+---
```
Comment (bowenli86) [inline docs/dev/table/hive/index.zh.md]: should the file name be 'index_zh.md'?

File: docs/dev/table/hive/read_write_hive.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+Using the `HiveCatalog`, Flink read and write from Hive Tables as an alternative to Hive's batch engine. Be sure to follow the instructions to include the correct [dependencies]({{ site.baseurl }}/dev/table/hive/#depedencies) in your application.
```
Comment (bowenli86) [inline docs/dev/table/hive/read_write_hive.md]:
```suggestion
Using the `HiveCatalog` and Flink's connector to Hive, Flink can read and write from Hive data as an alternative to Hive's batch engine. Be sure to follow the instructions to include the correct [dependencies]({{ site.baseurl }}/dev/table/hive/#depedencies) in your application. 
```

File: docs/dev/table/catalog.md [hunk 5]
```diff
@@ -23,344 +23,151 @@ specific language governing permissions and limitations
...
+### GenericInMemoryCatalog
 
-APIs are defined in `Catalog` interface. The interface defines a set of APIs to read and write catalog meta-objects such as database, tables, partitions, views, and functions.
+Flink sessions always have a built-in `GenericInMemoryCatalog` named `default_catalog`, which has a built-in default database named `default_database`.  All temporary metadata, such tables defined using `TableEnvironment#registerTable` is registered to this catalog. 
 
+### HiveCatalog
 
-Catalog Meta-Objects Naming Structure
--------------------------------------
+The `HiveCatalog` serves two purposes; as persistent storage for pure Flink metadata, and as an interface for reading and writing Hive tables. The Flink's [Hive documentation]({{ site.baseurl }}/dev/table/hive/index.html) provides full details on setting up the catalog and interfacing with an existing Hive installation.
```
Comment (bowenli86) [inline docs/dev/table/catalog.md]:
```suggestion
The `HiveCatalog` serves two purposes; as persistent storage for pure Flink metadata, and as an interface for reading and writing existing Hive metadata. The Flink's [Hive documentation]({{ site.baseurl }}/dev/table/hive/index.html) provides full details on setting up the catalog and interfacing with an existing Hive installation.
```

File: docs/dev/table/catalog.md [hunk 6]
```diff
@@ -23,344 +23,151 @@ specific language governing permissions and limitations
...
-Catalogs are pluggable. Users can develop custom catalogs by implementing the `Catalog` interface, which defines a set of APIs for reading and writing catalog meta-objects such as database, tables, partitions, views, and functions.
-
-Catalog Registration
---------------------
-
-## Register Catalog in Table API
-
-To register a catalog in Table API, users can create a catalog instance and register it through `TableEnvironment.registerCatalog(name, catalog)`.
-
-## Register Catalog in SQL CLI
```
Comment (bowenli86) [inline docs/dev/table/catalog.md]: we should keep this section somewhere, its last two paragraphs are very important information

File: docs/dev/table/hive/index.zh.md (hunk 3 above)
Comment (sjwiesman) [reply docs/dev/table/hive/index.zh.md]: no, its filename.zh.md

File: docs/dev/table/catalog.md (hunk 6 above)
Comment (sjwiesman) [reply docs/dev/table/catalog.md]: Done, moved to the user-defined catalog section

**Comment by sjwiesman** [comment]
Thanks for taking a look so quickly, I've addressed all of your feedback.

**Pull request closed without merging**